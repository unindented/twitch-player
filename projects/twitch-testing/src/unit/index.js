import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
