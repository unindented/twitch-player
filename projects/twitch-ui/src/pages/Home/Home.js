import React, { memo } from "react";
import { View } from "react-native";
import Heading from "../../components/Heading";

const Home = memo(({ testID = "home-page" }) => (
  <View testID={testID}>
    <Heading level="2">Home</Heading>
  </View>
));

export default Home;
