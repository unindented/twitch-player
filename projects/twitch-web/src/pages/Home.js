import Heading from "@twitch-player/ui/dist/components/Heading";
import React from "react";

const Home = () => <Heading level="2">Home</Heading>;

export default Home;
