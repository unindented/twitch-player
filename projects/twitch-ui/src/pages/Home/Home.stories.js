import { storiesOf } from "@storybook/react";
import React from "react";
import Home from ".";

storiesOf("Pages|Home", module).add("default", () => <Home />);
