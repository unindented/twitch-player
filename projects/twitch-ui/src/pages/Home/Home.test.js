import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import Home from "./Home";

describe("Home", () => {
  let instance;

  beforeEach(() => {
    instance = render(<Home />);
  });

  it("renders the page", () => {
    expect(instance.getByTestId("home-page")).toMatchSnapshot();
  });
});
