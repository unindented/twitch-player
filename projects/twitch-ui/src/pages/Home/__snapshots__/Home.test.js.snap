// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the page 1`] = `
<div
  class="css-view-1dbjc4n r-WebkitOverflowScrolling-150rngu r-flexDirection-eqz5dr r-flexGrow-16y2uox r-flexShrink-1wbh5a2 r-overflowX-11yh6sk r-overflowY-1rnoaur r-transform-1sncvnh"
  data-testid="home-page"
>
  <div
    class="css-view-1dbjc4n"
  >
    <mock-featured-channels
      heading="Featured Channels"
    />
    <mock-top-categories
      heading="Top Categories"
    />
    <mock-top-channels
      heading="Top Live Channels"
    />
    <mock-top-channels
      heading="Top PS4 Channels"
      platformtype="ps4"
    />
    <mock-top-channels
      heading="Top Xbox One Channels"
      platformtype="xbox"
    />
  </div>
</div>
`;
